### PR TITLE
MDX: Handle `<Story>` name starting with number

### PR DESCRIPTION
**Pull request by shilman**
Issue: #8467 

## What I did

Add underscore to exported var when story name starts with a number to fix JS parsing issues.

## How to test

See updated test case

### CODE REVIEW

File: addons/docs/src/mdx/mdx-compiler-plugin.js [hunk 1]
```diff
@@ -19,12 +19,16 @@ function getAttr(elt, what) {
...
 
 const isReserved = name => RESERVED.exec(name);
+const startsWithNumber = name => /^\d/.exec(name);
 
 const sanitizeName = name => {
   let key = camelCase(name);
   if (isReserved(key)) {
     key = `${key}Story`;
   }
+  if (startsWithNumber(key)) {
```
Comment (tmeasday) [inline addons/docs/src/mdx/mdx-compiler-plugin.js]: Probably put this before the `isReserved` check?

**Review by tmeasday — CHANGES_REQUESTED** [review]
LGTM, one minor nitpick

File: addons/docs/src/mdx/mdx-compiler-plugin.js (hunk 1 above)
Comment (shilman) [reply addons/docs/src/mdx/mdx-compiler-plugin.js]: why?

Comment (shilman) [reply addons/docs/src/mdx/mdx-compiler-plugin.js]: fixed

Comment (tmeasday) [reply addons/docs/src/mdx/mdx-compiler-plugin.js]: Add "Story" to the end of a name is never going to make it start with a number.

Adding `_` to the start of a name is probably never going to make it a reserved word, but I don't know off the top of my head what all the reserved words are so it seems safer that way round.

**Pull request merged**